lib/testScenario/apiScenarioYamlLoader.ts

```diff
@@ -0,0 +1,145 @@
+import * as fs from "fs";
+import { resolve as pathResolve, dirname } from "path";
+import { JSONPath } from "jsonpath-plus";
+import { injectable } from "inversify";
+import { Type, YAMLException, load as yamlLoad, DEFAULT_SCHEMA } from "js-yaml";
+import { default as AjvInit, ValidateFunction } from "ajv";
+import { Loader } from "../swagger/loader";
+import { FileLoader } from "../swagger/fileLoader";
+import { RawScenarioDefinition } from "./apiScenarioTypes";
+import { ApiScenarioDefinition } from "./apiScenarioSchema";
+
+const ajv = new AjvInit({
+  useDefaults: true,
+});
+@injectable()
+export class ApiScenarioYamlLoader implements Loader<RawScenarioDefinition> {
+  private fileCache: Map<string, string> = new Map();
+  private validateApiScenarioFile: ValidateFunction;
+
+  public constructor(private fileLoader: FileLoader) {
+    this.validateApiScenarioFile = ajv.compile(ApiScenarioDefinition);
+  }
+
+  public async load(filePath: string): Promise<RawScenarioDefinition> {
+    this.fileCache.clear();
+
+    const fileContent = await this.fileLoader.load(filePath);
+    yamlLoad(fileContent, {
+      schema: DEFAULT_SCHEMA.extend(
+        new Type("!include", {
+          kind: "scalar",
```

Comment:
what does the kind 'scalar' mean? better to add doc link about howt to add customize extension for yaml.

Reply:
It's data type of YAML standard, not supposed to be used by external users.